### ISSUE TITLE
Private Window turns off DNS over HTTPS and DNS requests are send in clear

### ISSUE THREAD

**Issue by SkyperTHC** [brave/brave-browser#7379]
## Description 

File -> New Private Window.
Brave sends DNS requests out in clear even when DoH is configured and DoH is working on 'normal' (non-private) windows.

## Steps to Reproduce

   1.  Use brave normally in a normal window. Observe DNS requests are send out via DoH.
   2. File -> New Private Window. Go to any URL. Observe DNS requests are send in clear (udp/53) and not via DoH.

## Actual result:

DNS requests are send in clear even when DoH is configured.

## Expected result:

DNS requests to be send out via DoH while in 'Private Windows’. 

## Reproduces how often: 

Every time.

## Brave version (brave://version info)
Version 1.1.20 Chromium: 79.0.3945.74 (Official Build) (64-bit)

## Version/Channel Information:

- Can you reproduce this issue with the current release? LATEST stable
- Can you reproduce this issue with the beta channel? no
- Can you reproduce this issue with the dev channel? no
- Can you reproduce this issue with the nightly channel? no

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? not tried. Unrelated?
- Does the issue resolve itself when disabling Brave Rewards? not tried. Unrelated?
- Is the issue reproducible on the latest version of Chrome? not tried.

## Miscellaneous Information:

N/A

**Comment by SkyperTHC** [comment]
anyone?

**Comment by rebron** [comment]
@tomlowenthal @jumde Can you guys comment/triage this one?

**Comment by fmarier** [comment]
I just verified the latest Nightly (1.6) on Windows and it does use DoH for DNS requests both in normal and private windows when Cloudflare is set as the OS resolver.

Here's how I tested this:

1. Open Brave.
2. Visit https://1.1.1.1/help to confirm that DoH is enabled.
3. Start a wireshark capture.
4. Visit a bunch of random websites to trigger DNS lookups.
5. Stop the wireshark capture.
6. Sort the requests by type ("DNS" is at the top) and look for any DNS lookups for the websites visited in step 4.
